gcp-observability/src/main/java/io/grpc/gcp/observability/GcpObservability.java

```diff
@@ -117,6 +122,16 @@ public void close() {
         throw new IllegalStateException("GcpObservability already closed!");
       }
       sink.close();
+      if (config.isEnableCloudMonitoring() || config.isEnableCloudTracing()) {
+        try {
+          // Sleeping before shutdown to ensure all metrics and traces are flushed
+          Thread.sleep(
+              TimeUnit.MILLISECONDS.convert(2 * METRICS_EXPORT_INTERVAL, TimeUnit.SECONDS));
+        } catch (InterruptedException e) {
+          Thread.currentThread().interrupt();
+          logger.log(Level.SEVERE, "Caught exception during sleep", e);
```

Comment:
Shouldn't the logging happen before `interrupt()`? The `interrupt()` itself breaks the flow of control so the log statement may never get executed.

I understand this was backported from master but may need to be fixed there as well.